### ISSUE TITLE
testing: go test doen't report a test's name if it times out 

### ISSUE THREAD

**Issue by distinctdan** [golang/go#50542]
### What version of Go are you using (`go version`)?
Latest on playground, looks like 1.17.6

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?
Playground

### What did you do?
One of my tests is failing because it's waiting on a channel. When I run `go test` it does timeout, but it doesn't give me the name of the failing test. We have close to 1000 tests, so it was frustrating to track down which test is causing the problem. Here's a playground link that demonstrates the issue: https://go.dev/play/p/7ruH91WlBxd

### What did you expect to see?
Go should report the name of a failing test, even if the failure is from a timeout

### What did you see instead?
No test name

**Comment by bcmills** [comment]
Duplicate of #39038
